### DISCUSSION TITLE
How to delete a Report with syntax error in it

### DISCUSSION THREAD

**Discussion by kawalex** [netbox-community/netbox#12841] [Help Wanted!] [answered]
Hi,
On Netbox v3.5.3, when a report is added or updated with a syntax error in it, the main report page (via Customization/Reports) is inaccessible and a "Server Error' page is displayed instead.

As the Report page is not reachable, the report cannot be deleted or corrected.

What are the correct actions to do to solve this issue ?

Thanks !

**Comment by candlerb** [comment]
In some ways scripts handle this better (they don't cause a server error), but worse (they don't show you what line the error was in). See #12766.

The problem with reports is, if you move them out of the way:

```
cd /opt/netbox/netbox/reports
mv broken_report.py broken_report.py.xxx
```

... then you get a different Server Error exception because it still tries to open the (now non-existent) file:

```
<class 'FileNotFoundError'>

[Errno 2] No such file or directory: '/opt/netbox/netbox/reports/broken_report.py'
```

The only solution I can find is a manual SQL update:

```
sudo -u postgres psql netbox
delete from core_managedfile where file_root='reports' and file_path='broken_report.py';
```

(Dropping the corrected report back into the directory doesn't make it reappear though. AFAICS, you have to re-upload it via the web interface)

**Comment by candlerb** [comment, marked as answer]
A simpler solution: replace the broken report with a zero-byte file.  This avoids having to touch the database.

```
cd /opt/netbox/netbox/reports
mv broken_report.py broken_report.py.xxx
touch broken_report.py
```

Then you can delete it from the GUI.

Also raised as #12842

**Reply by kawalex** [reply]
Thanks ! Indeed this is the only "simple" solution I've found...
